tests/unit/test_charm.py

```diff
@@ -3,6 +3,151 @@
 #
 # Learn more about testing at: https://juju.is/docs/sdk/testing
 
+import ops
+import pytest
+from ops import testing
+from unittest.mock import patch, PropertyMock
+from src.charm import CassandraCharm
+from src.core.state import PEER_RELATION
+
+
+def test_start_maintenance_status_when_starting():
+    """Charm enters MaintenanceStatus when Cassandra is not yet healthy."""
+    ctx = testing.Context(CassandraCharm)
+    relation = testing.PeerRelation(id=1, endpoint=PEER_RELATION)
+    state_in = testing.State(leader=True, relations={relation})
+
+    with (
+        patch("managers.config.ConfigManager.render_env"),
+        patch("workload.CassandraWorkload.start"),
+        patch("subprocess.run"),
+    ):
+        state_out = ctx.run(ctx.on.start(), state_in)
+        assert state_out.unit_status == ops.MaintenanceStatus("waiting for Cassandra to start")
+        assert state_out.get_relation(1).local_unit_data.get("workload_state") == 'active'
```

Comment:
`cluster_state` becomes `active` along with `workload_state` if unit is leader, so we can also check it here.
```suggestion
        assert state_out.get_relation(1).local_unit_data.get("workload_state") == 'active'
        assert state_out.get_relation(1).local_app_data.get("cluster_state") == 'active'
```